hooks/honeybadger/honeybadger.go

```diff
@@ -0,0 +1,36 @@
+package logrus_honeybadger
+
+import (
+	"github.com/DavidHuie/gobadger"
+	"github.com/Sirupsen/logrus"
+)
+
+// HoneybadgerHook to send exceptions to an exception-tracking service compatible
+// with the Honeybadger API.
+type HoneybadgerHook struct {
+	conn *gobadger.Conn
+}
+
+func NewHoneyBadgerHook(apiKey string) *HoneybadgerHook {
+	return &HoneybadgerHook{gobadger.NewConn(apiKey)}
+}
+
+func (hook *HoneybadgerHook) Fire(entry *logrus.Entry) error {
+	err := hook.conn.Error(entry.Message)
```

Comment:
What do you think about something like this?

```
func (hook *HoneybadgerHook) Fire(entry *logrus.Entry) error {
	doneChan := make(chan struct{})

	go func(doneChan chan struct{}) {
		err := hook.conn.Error(entry.Message)
		if err != nil {
			entry.Logger.WithFields(logrus.Fields{
				"source": "honeybadger",
				"error":  err,
			}).Warn("Failed to send error to HoneyBadger")
		}
		doneChan <- struct{}{}
	}(doneChan)

	select {
	case <-time.After(5 * time.Second):
		message := "Timed out connecting to HoneyBadger"
		entry.Logger.WithFields(logrus.Fields{
			"source": "honeybadger",
			"error":  errors.New(message),
		}).Warn(message)
	case <-doneChan:
	}

	return nil
}
```

It doesn't look like the honeybadger client that I am using support any kind of timeout.

If you think that the slowness on the hooks is critical, perhaps the timeout should be implemented on the hook call and not inside the hook?